crates/resolver/src/config.rs

```diff
@@ -236,17 +238,40 @@ impl Default for ResolverConfig {
 #[cfg_attr(feature = "serde-config", derive(Serialize, Deserialize))]
 pub enum Protocol {
     /// UDP is the traditional DNS port, this is generally the correct choice
+    #[cfg_attr(feature = "serde-config", serde(alias = "udp", alias = "UDP"))]
     Udp,
     /// TCP can be used for large queries, but not all NameServers support it
+    #[cfg_attr(feature = "serde-config", serde(alias = "tcp", alias = "TCP"))]
     Tcp,
     /// Tls for DNS over TLS
     #[cfg(feature = "dns-over-tls")]
+    #[cfg_attr(
```

Comment:
I feel like this is overkill... What's the use case here? I think it'd make more sense to do a `rename_all("kebab-case")` on the enum here and not bother with all the alternative spellings.

If you really want this level of customizability, just write your own `NameServerConfig` with an `Into` impl.

Reply:
Yup. I removed those unnecessary aliases.